server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/executeBash.ts

```diff
@@ -536,6 +536,24 @@ export class ExecuteBash {
         })
     }
 
+    /**
+     * Re‑creates the raw bytes from the received string (Buffer.from(text, 'binary')).
+     * Detects UTF‑16 LE by checking whether every odd byte in the first 32 bytes is 0x00.
+     * Decodes with buf.toString('utf16le') when the pattern matches, otherwise falls back to UTF‑8.
```

Comment:
is this actually a reasonable assumption? isnt the encoding defined by the user's shell locale?

Reply:
`cmd.exe /u` forces built‑ins to emit `UTF‑16LE`, bypassing whatever code page the user’s console is set to.

I'm not sure if true “universal” detection is practical. This heuristic covers the dominant cases (`UTF‑16LE` and `UTF‑8`).